pallets/xyk/src/lib.rs

```diff
@@ -512,10 +535,366 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		#[pallet::weight(10_000)]
+		pub fn claim_rewards(
+			origin: OriginFor<T>,
+			liquidity_token_id: TokenId,
+			amount: Balance,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			<Self as XykFunctionsTrait<T::AccountId>>::claim_rewards(
+				sender,
+				liquidity_token_id,
+				amount,
+			)?;
+
+			Ok(().into())
+		}
 	}
 }
 
 impl<T: Config> Pallet<T> {
+	pub fn calculate_available_rewards_for_pool(
+		liquidity_asset_id: TokenId,
+		block_number: u32,
+	) -> Result<Balance, DispatchError> {
+		let mut already_claimed_pool: Balance = 0;
+		if LiquidityMiningPoolClaimed::<T>::contains_key(&liquidity_asset_id) {
+			already_claimed_pool =
+				Self::liquidity_mining_pool_claimed(&liquidity_asset_id).unwrap();
```

Comment:
again if you change the storage type to non-Option you wouldnt have to unwrap here ;)